### PR TITLE
Moved Math functions from render to Utils

### PR DESCRIPTION
**Pull request by Raelr**
In an attempt to limit our dependency on `glm`, I've started moving all the math functions from the renderer to utils and have implemented most of them.

### CODE REVIEW

**Review by J-Mo63 — APPROVED** [review]
Looks fine, just a bit confused why those helper libs should be classes rather than namespaces like `FloatUtils`.

File: engine/core/entity/Entity.cpp [hunk 1]
```diff
@@ -88,7 +88,8 @@ void Entity::SetPosition(const Vec3& position)
 
 void Entity::SetRotation(float rotation)
 {
-    transform.SetRotation(fmod(rotation, 360.f));
+    transform.SetRotation(
```
[the remark this answers is not in the document]
Comment (Raelr) [reply engine/core/entity/Entity.cpp:91]: This does exist in the Xform class:

```c++
    void SetRotation(const Vec3& newRotation)
    {
        rotation = newRotation;
    }
```

File: engine/core/TransitionAdapter.h [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #include <raylib/Vector3.hpp>
 
 #include "../utils/Colour.h"
-#include "../utils/Maths.h"
+#include "utils/math/Maths.h"
```
[the remark this answers is not in the document]
Comment (Raelr) [reply engine/core/TransitionAdapter.h]: There were a few of these lying around. Not sure why. I thought I found all the cases but I guess I missed one. This is now fixed again.

File: engine/utils/math/Float.h [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+#endif
+
+namespace Siege
+{
+/**
+ * A library for computing common float computations.
+ */
+class Float
+{
+public:
```
[the remark this answers is not in the document]
Comment (Raelr) [reply engine/utils/math/Float.h]: It doesn't. I guess I copied this convention from other projects I saw. I reverted this back to a namespace to retain consistency.

File: engine/utils/math/Graphics.h [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+#include "mat/Mat4.h"
+
+namespace Siege
+{
+/**
+ * A collection of functions for calculating common graphics related operations.
+ */
+class Graphics
+{
+public:
```
[the remark this answers is not in the document]
Comment (Raelr) [reply engine/utils/math/Graphics.h]: See above.

**Pull request merged**